drivers/amazonec2/amazonec2.go

```diff
@@ -269,12 +270,19 @@ func (d *Driver) Create() error {
 	}
 
 	d.InstanceId = instance.InstanceId
+	d.IPAddress = instance.IpAddress
+
+	if len(instance.NetworkInterfaceSet) > 0 {
+		d.PrivateIPAddress = instance.NetworkInterfaceSet[0].PrivateIpAddress
+	}
 
 	d.waitForInstance()
 
-	log.Debugf("created instance ID %s, IP address %s",
+	log.Debugf("created instance ID %s, IP address %s, Private IP address %s",
 		d.InstanceId,
-		d.IPAddress)
+		d.IPAddress,
+		instance.NetworkInterfaceSet[0].PrivateIpAddress,
```

Comment:
Let's use `d.PrivateIPAddress` here instead - it is less verbose and will default to empty string if this field is not present in the response.